### PR TITLE
feat(models): add IREQ model and related documentation (ISO 11079)

### PR DESCRIPTION
**Pull request by LianqiWang111**
### Description

This pull request introduces the implementation of the IREQ (Required Clothing Insulation) model according to ISO 11079.
The update includes:

- A new model file **pythermalcomfort/models/ireq.py** implementing IREQ calculations.
- Updated class definitions in **classes_input.py** and **classes_return.py** to support the new model.
- Added unit tests in **tests/test_ireq.py** covering scalar, vectorized, and broadcasting input cases.
- Updated documentation in **docs/documentation/models.rst**.
- Updated **CHANGELOG.rst**.

### Type of change

 New feature (non-breaking change which adds functionality)
 This change requires a documentation update

### How Has This Been Tested?

All tests for the calc_ireq() function were conducted in tests/test_ireq.py using pytest.

Test A: Verified scalar input calculations to ensure numerical stability and physically valid outputs.

Test B: Verified vectorized and broadcasting input handling with NumPy arrays to ensure consistent output shapes and array compatibility.

**Test Configuration:**

Firmware version: N/A
Hardware: macOS (M1)
Toolchain: Python 3.11, pytest 8.4.1
SDK: N/A

### Checklist

- My code follows the style guidelines of this project
- I have performed a self-review of my code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

## Summary by CodeRabbit

* **New Features**
  * Added IREQ (ISO 11079:2007) calculations producing IREQ, ICL and DLE results with structured, vectorised outputs and new input fields (duration, t_re, t_sk) and input types.

* **Bug Fixes / Quality**
  * Strengthened input validation, broadcasting, numeric‑stability and finite‑value checks.

* **Documentation**
  * Added user documentation for the IREQ model and outputs.

* **Tests**
  * Comprehensive tests for scalar/array inputs, broadcasting, edge cases and invalid inputs.

### CODE REVIEW

File: tests/test_ireq.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    # String input should raise a ValueError
+    with pytest.raises(ValueError):
+        calc_ireq(
+            m="abc", w_work=0, tdb=0, tr=0, p_air=2, v_walk=0.5, v=0.5, rh=50, clo=1.0
+        )
+
+    # Missing required parameters should raise a TypeError
+    with pytest.raises(TypeError):
+        calc_ireq()
+
```
Comment (KristinaM93) [inline tests/test_ireq.py:65]: Add a test case for p_air <= 0 to ensure the model raises a ValueError for log-domain issues. This is essential for numerical safety.

File: tests/test_ireq.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            tdb=tdb,
+            tr=tr,
+            p_air=8.0,
+            v_walk=v,
+            v=v,
+            rh=85,
+            clo=1.0,  # CLO doesn't affect IREQ_neutral directly in comparison
+        )
+
+        assert_allclose(result.ireq_neutral, ireq_ref, rtol=0.1)
```
Comment (KristinaM93) [inline tests/test_ireq.py:113]: Maybe include a test for the DLE “more than 8” output path. Ensures that the function correctly returns strings when exposure exceeds 8 hours.

**Review by KristinaM93 — COMMENTED** [review]
Just some suggestions for test_ireg.py
